doc/dynamo0p3.rst

```diff
@@ -1049,6 +1050,41 @@ spaces associated with the arguments listed in ``meta_args``.  In this
 case we require both for the W0 function space but only basis
 functions for W1.
 
+meta_reference_element
+######################
+
+A kernel that requires properties of the reference element specifies
+those properties through the ``meta_reference_element`` metadata
+entry.  (If no reference element properties are required then this
+metadata should be omitted.)  Consider the following example kernel
+metadata::
+
+  type, extends(kernel_type) :: testkern_type
+    type(arg_type), dimension(2) :: meta_args = &
+        (/ arg_type(gh_field, gh_read, w1),     &
+	   arg_type(gh_field, gh_write, w0) /)
+    type(reference_element_data_type), dimension(2) :: &
+      meta_reference_element =                         &
+        (/ mesh_data_type(number_horizontal_faces),    &
+	   mesh_data_type(normal_to_face ) /)
+  contains
+    procedure, nopass :: code => testkern_code
+  end type testkern_type
+
+This metadata specifies that the ``testkern_type`` kernel requires two
+properties of the reference element. The supported properties are
+listed below:
+
+======================= ===================================================
+Name                    Description
+======================= ===================================================
+normal_to_face          Array of normals for each horizontal face indexed
+                        as (face, component).
+number_horizontal_faces Number of horizontal faces on the reference element.
+out_face_normal         Array of normals for each horizontal face indexed
+                        as (component, face).
+======================= ===================================================
+
```

Comment:
We need a section for the `meta_mesh` as well. My suggestion is below.

```
meta_mesh
#########

A kernel that requires properties of the LFRic mesh object specifies
those properties through the ``meta_mesh`` metadata entry.  (If no
mesh properties are required then this metadata should be omitted.)
Consider the following example kernel metadata::

  type, extends(kernel_type) :: testkern_type
    type(arg_type), dimension(2) :: meta_args = &
        (/ arg_type(gh_field, gh_read, w1),     &
	   arg_type(gh_field, gh_write, w0) /)
    type(mesh_data_type), dimension(1) ::       &
      meta_mesh =                               &
        (/ mesh_data_type(adjacent_face) /)
  contains
    procedure, nopass :: code => testkern_code
  end type testkern_type

This metadata specifies that the ``testkern_type`` kernel requires one
property of the mesh. There is currently pne supported property:

======================= ===================================================
Name                    Description
======================= ===================================================
adjacent_face           Local ID of a neighbouring face in each adjacent
                        cell indexed as (face).
======================= ===================================================
```

Reply:
Thanks but that's a different PR :-)